remoteappmanager/db/orm.py

```diff
@@ -129,6 +130,54 @@ def reset(self):
         Base.metadata.create_all(self.engine)
 
 
+class AccountingInterface(ABCAccounting):
+
+    def __init__(self, url, **kwargs):
+        self.db = Database(url, **kwargs)
+
+        # We keep the same session for all transactions
+        # so that the session remains on one thread
+        self.session = self.db.create_session()
```

Comment:
You are supposed to create a session and do multiple transactions with them, potentially, but in our little application, it's probably a good idea to start the session, start the transaction, complete the transaction and never reuse the session again.

so one thing that you could do, is to have a begin and end methods in your database application. In the sqlalchemy one it creates the session, in the csv one it does nothing.